### PR TITLE
Fix `VisionEncoderDecoder` Positional Arg

### PR DESCRIPTION
**Pull request by nickthegroot**
# What does this PR do?

When using certain models as the encoder for `VisionEncoderDecoder` (e.g. `LayoutLMv3`), `pixel_values` could incorrectly be the model's first positional argument (`input_ids` for `LayoutLMv3`).

This PR fixes the issue by specifying `pixel_values` as a keyword argument and adds the appropriate tests to validate going forward.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

## Who can review?

@amyeroberts

### CODE REVIEW

File: tests/models/vision_encoder_decoder/test_modeling_vision_encoder_decoder.py [hunk 1]
```diff
@@ -679,6 +681,118 @@ def prepare_config_and_inputs(self):
...
+        enc_dec_model = VisionEncoderDecoderModel(encoder=encoder_model, decoder=decoder_model)
+        enc_dec_model.to(torch_device)
+        outputs_encoder_decoder = enc_dec_model(
+            input_ids=input_ids,
+            pixel_values=pixel_values,
+            **kwargs,
+            decoder_input_ids=decoder_input_ids,
+            decoder_attention_mask=decoder_attention_mask,
+            output_attentions=True,
+        )
```
Comment (amyeroberts) [inline tests/models/vision_encoder_decoder/test_modeling_vision_encoder_decoder.py:740]: nit 

```suggestion
        outputs_encoder_decoder = enc_dec_model(
            input_ids=input_ids,
            pixel_values=pixel_values,
            decoder_input_ids=decoder_input_ids,
            decoder_attention_mask=decoder_attention_mask,
            output_attentions=True,
            **kwargs,
        )
```

**Review by amyeroberts — APPROVED** [review]
Great - thanks for fixing and adding tests! ❤️ 

Just some small nits, otherwise good to merge :)

**Comment by nickthegroot** [comment]
@amyeroberts fixed in 69c2385 & ready to merge!

**Comment by amyeroberts** [comment]
@nickthegroot Code quality checks are currently failing - can you run `make fixup` and push the changes?

**Comment by nickthegroot** [comment]
> 406469f

@amyeroberts - whoops, fixed in 406469f!

**Pull request merged**